packages/graph-explorer/src/core/StateProvider/displayEdge.ts

```diff
@@ -0,0 +1,154 @@
+import { Edge, EdgeId, VertexId } from "@/@types/entities";
+import { selector, selectorFamily, useRecoilValue } from "recoil";
+import { textTransformSelector } from "@/hooks/useTextTransform";
+import {
+  edgeTypeConfigSelector,
+  vertexTypeAttributesSelector,
+  vertexTypeConfigSelector,
+} from "../ConfigurationProvider/useConfiguration";
+import { edgesAtom, edgesSelectedIdsAtom } from "./edges";
+import {
+  DisplayAttribute,
+  getSortedDisplayAttributes,
+} from "./displayAttribute";
+import {
+  DisplayEdgeTypeConfig,
+  displayEdgeTypeConfigSelector,
+} from "./displayTypeConfigs";
+import { queryEngineSelector } from "../connector";
+
+/** Represents an edge's display information after all transformations have been applied. */
+export type DisplayEdge = {
+  entityType: "edge";
+  id: EdgeId;
+  displayId: string | null;
+  displayTypes: string;
+  typeConfig: DisplayEdgeTypeConfig;
+  source: EdgeVertex;
+  target: EdgeVertex;
+  attributes: DisplayAttribute[];
+};
+
+type EdgeVertex = {
+  id: VertexId;
+  displayId: string;
+  displayTypes: string;
+};
+
+/** Maps all `Edge` instances in the graph canvas to `DisplayEdge` instances. */
+export function useDisplayEdgesInCanvas() {
+  return useRecoilValue(displayEdgesInCanvasSelector);
+}
+
+/** Finds the `DisplayEdge` instance for a give `EdgeId` within the edges added to the graph canvas. */
+export function useDisplayEdge(id: EdgeId) {
+  const edge = useRecoilValue(displayEdgesInCanvasSelector).get(id);
+
+  if (!edge) {
+    throw new Error(`Edge with id ${id} not found in displayEdges`);
+  }
+
+  return edge;
+}
+
+/** Maps a `Edge` instance to a `DisplayEdge` instance using the schema and any user preferences. */
+export function useDisplayEdgeFromEdge(edge: Edge) {
+  return useRecoilValue(displayEdgeSelector(edge));
+}
+
+/** Maps all `Edge` instances which are selected in the graph canvas to `DisplayEdge` instances. */
+export function useSelectedDisplayEdges() {
+  const selectedIds = useRecoilValue(edgesSelectedIdsAtom);
+  const displayEdges = useDisplayEdgesInCanvas();
```

Comment:
Nit: if there is only one selected edge but many displayed edges, is this doing unnecessary work to iterate through all of the displayed edges when only the selected one needs to be mapped?

Reply:
Fair point. Technically, all the display nodes and edges will be created already, but this code doesn't need to force that behavior.